cmd/cluster-network-operator/main.go

```diff
@@ -69,49 +68,32 @@ func main() {
 		os.Setenv("KUBERNETES_SERVICE_PORT", url.Port())
 	}
 
-	// Get a config to talk to the apiserver
-	cfg, err := config.GetConfig()
-	if err != nil {
-		log.Fatal(err)
+	if err := command.Execute(); err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
 	}
+}
 
-	// become leader
-	err = BecomeLeader(context.TODO(), LOCK_NAME)
-	if err != nil {
-		log.Fatal(err)
+func newNetworkOperatorCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "network-operator",
+		Short: "Openshift Cluster Network Operator",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.Help()
+			os.Exit(1)
+		},
 	}
 
-	// Create a new Cmd to provide shared dependencies and start components
-	mgr, err := manager.New(cfg, manager.Options{
-		Namespace:          namespace,
-		MapperProvider:     k8sutil.NewDynamicRESTMapper,
-		MetricsBindAddress: "0",
-	})
-	if err != nil {
-		log.Fatal(err)
-	}
+	cmdcfg := controllercmd.NewControllerCommandConfig("network-operator", version.Get(), operator.RunOperator)
 
-	log.Print("Registering Components.")
-	if err := operv1.Install(mgr.GetScheme()); err != nil {
-		log.Fatal(err)
-	}
-	if err := configv1.Install(mgr.GetScheme()); err != nil {
-		log.Fatal(err)
-	}
-	if err := netopv1.Install(mgr.GetScheme()); err != nil {
-		log.Fatal(err)
-	}
+	// We'll use our own leader election
+	cmdcfg.DisableLeaderElection = true
```

Comment:
The configmap name isn't customizable, so we can't guarantee a seamless upgrade, since the lock key name changes.

Given that multiple instances of the CNO are extremely unlikely, we could swap the key and assume this is safe.